### DISCUSSION TITLE
Flexible authentication options - username or mobile number, login links and Firebase/Supabase integration

### DISCUSSION THREAD

**Discussion by anandfursule** [directus/directus#21892] [Feature Requests]
### Summary

I prupose adding enhanced authentication options to support username or mobile number login alongside email id. This update would also include support for login links and integration with Firebase and Supabase.

### Basic Example

Currently, our Directus-based multitenant app only supports email and password authentication. With this feature request implemented, users would have the option to log in using either their username or mobile number, along with their password. Additionally, the system would generate login links for seamless access and integrate seamlessly with Firebase and Supabase for enhanced security and functionality.

### Motivation

Our app caters to diverse user preferences and needs. By expanding the authentication options to include usernames, mobile numbers, and login links, we can provide a more user-friendly and secure experience. Integration with Firebase and Supabase further enhances our app's capabilities, ensuring robust authentication mechanisms that meet modern security standards.

### Detailed Design

1. User Interface Changes
2. Backend Changes
3. Integration with Firebase and Supabase
4. User Experience Enhancements

### Requirements List

Must-Have (Must)
1. Allow users to log in using username or mobile number alongside passwords.
2. Generate secure login links for seamless access.
3. Integrate with Firebase and Supabase for authentication services.

Should-Have (Should)
1. Support configurable login options (username, mobile number, or email).
2. Provide options for password recovery and reset.
3. Implement multi-factor authentication (MFA) for added security.

Could-Have (Could)
1. Enable social media authentication options (e.g., login with Google, Facebook).
2. Implement passwordless authentication (e.g., OTP-based login).
3. Allow users to manage their authentication preferences in the user profile settings.

Won't-Have (Won't)
1. Support legacy authentication methods not aligned with modern security standards.
2. Integrate with proprietary authentication systems without standard API support.
3. Implement biometric authentication features (e.g., fingerprint, facial recognition) in the initial release.

### Drawbacks

1. Complexity and Development Time
2. Maintenance Overhead

### Alternatives

None

### Adoption Strategy

Haven't thought about it

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
